discord_thread_summarizer_using_openai/main.bal

```diff
@@ -0,0 +1,79 @@
+import ballerina/io;
+import ballerina/time;
+
+const string DISCORD_API_URL = "https://discord.com/api/v9";
+const TYPE_FORUM_CHANEL = 15;
+
+configurable string botToken = ?;
+configurable int forumChannelId = ?;
+
+function readThreads(ActiveThreads activeThreads) returns error? {
+
+    foreach ChannelThread thread in activeThreads.threads {
+
+        // check if the thread has any recent messages
+        if check checkRecentMessages(thread) {
+            string prompt = check constructPrompt(thread);
+            io:println(prompt);
+
+            string|error generateChatCompletionResult = generateChatCompletion(prompt, model = "gpt-3.5-turbo");
+
+            if generateChatCompletionResult is string {
+                io:println(generateChatCompletionResult);
+            } else {
+                io:println("Error: " + generateChatCompletionResult.toBalString());
+            }
+            io:println("--------------------------------------------------\n\n");
+        }
+    }
+}
+
+function constructPrompt(ChannelThread thread) returns string|error {
+    string prompt = PROMPT + "Thread URL: " + getThreadURL(thread.guild_id, thread.id.toString()) + "\nTitle: " + thread.name.toString() + "\nQuestion: ";
+    Message[] allMessages = check getMessages(thread.id.toString());
+
+    // reverse the array so that the messages are in chronological order
+    allMessages = reverseArray(allMessages);
+
+    boolean firstMessage = true;
+    foreach Message message in allMessages {
+        string formattedTimestamp = message.timestamp.toString();
+        if (message.content == "") {
```

Comment:
```suggestion
        if message.content == "" {
```

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/1dbf76e067fd08cc0430182aa02b7c98d154f3ac